config: warn for unrecognized keys

CapacitorSet opened the pull request.

nl5887 commented on config/config.go:
```diff
@@ -50,6 +50,8 @@ var format = logging.MustStringFormatter(
 
 // Config defines the central type where all configuration is umarhsalled to.
 type Config struct {
+	TomlMetadata toml.MetaData
```
What about having Config embed toml.MetaData, like this:

```
type Config struct {
toml.MetaData
}
```

CapacitorSet replied:
Makes sense, I fixed it.

CapacitorSet commented:
While I was at it, I also added a commit to warn for unused channel. The first times I used Honeytrap it happened that I added channels and expected them to work "out of the box", and wondering why they didn't - a warning message would be useful.

nl5887 commented:
thanks!
